prow/pjutil/abort.go

```diff
@@ -17,19 +17,27 @@ limitations under the License.
 package pjutil
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"sort"
 	"strings"
 
 	jsonpatch "github.com/evanphx/json-patch"
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
 	ktypes "k8s.io/apimachinery/pkg/types"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 
 	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
 	"k8s.io/test-infra/prow/github/reporter"
 )
 
+// ctrlruntimeprowClient a minimalistic prow client required by the aborter
+type ctrlruntimeprowClient interface {
```

Comment:
nit: this name leaks the implementation details, it's just a patch client right?

Reply:
Updated. I should mention thought that unless there is an actual reason like tests that mock it, I find these subinterface definitions to be mostly a premature optimization

Reply:
I assumed we needed it for test mocks .. if we don't then we shouldn't bother (or we should write tests)